File: api/TimeTracker.Domain/Activity.cs
namespace TimeTracker.Domain
{
    public class Activity : BaseEntity
    {
        public Guid UserId { get; set; }
        public string Description { get; set; } = string.Empty;
        public List<Tag>? Tags { get; set; }
        public Project? Project { get; set; }
        public long StartInMilliseconds { get; set; }
        public long? EndInMilliseconds { get; set; }
    }
}

File: web-client/src/models/activity.ts
import { Project } from "./project";
import { Tag } from "./tag";

export interface ActivityListVm {
  activities: ActivityLookupDto[];
}

export interface ActivityLookupDto {
  id: string;
  description: string;
  tags: Tag[];
  project?: Project;
  startInMilliseconds: number;
  endInMilliseconds?: number;
}

export interface CreateActivityDto {
  description: string;
  tagIds: string[];
  projectId?: string;
  startInMilliseconds: number;
  endInMilliseconds?: number;
}

export interface StartActivityDto {
  description: string;
  tagIds: string[];
  projectId?: string;
}

export interface UpdateActivityDto {
  id: string;
  description: string;
  tagIds: string[];
  projectId?: string;
  startInMilliseconds: number;
  endInMilliseconds?: number;
}
